DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/stp/test_stp_with_traffic.py

```diff
@@ -0,0 +1,190 @@
+        'agreement': 1,
+        'forwarding': 1,
+        'learning': 1,
+        'portRole': PortRole.DESIGNATED.value,
+        'proposal': 1,
+        'protocol': '0027',
+    } if version == 'rstp' else {}
+    streams = {
+        f'{ports[0]} {version}': {
+            'type': 'bpdu',
+            'version': version,
+            'ep_source': ports[0],
+            'ep_destination': ports[0],
+            'rate': 1,  # pps
+            'allowSelfDestined': True,
+            'srcMac': root_mac,
+            **stp,
+            **rstp,
+        },
+        f'{ports[1]} {version}': {
+            'type': 'bpdu',
+            'version': version,
+            'ep_source': ports[1],
+            'ep_destination': ports[1],
+            'rate': 1,  # pps
+            'allowSelfDestined': True,
+            'srcMac': rand_mac,
+            **stp,
+            **rstp,
+            'bridgeIdentifier': f'8000{rand_mac.replace(":", "")}',
+        },
+        traffic: {
+            'type': 'ethernet',
+            'ep_source': ports[2],
+            'frame_rate_type': 'bps_rate',
+            'srcMac': get_rand_mac('00:11:XX:XX:XX:XX'),
+            'rate': rate_bps,
+        },
+        traffic_1: {
+            'type': 'ethernet',
+            'ep_source': ports[3],
+            'frame_rate_type': 'bps_rate',
+            'srcMac': get_rand_mac('00:22:XX:XX:XX:XX'),
+            'rate': rate_bps,
+        },
+    }
+    await tgen_utils_setup_streams(tgen_dev, None, streams)
+    await tgen_utils_start_traffic(tgen_dev)
+    # don't stop
+    await asyncio.sleep(convergence_time_s)
+
+    for _ in range(5):
+        # 6. Verify port 2 state is blocking
+        out = await Mstpctl.show(input_data=[{dent: [
+            {'parameter': 'portdetail',
+             'bridge': bridge,
+             'options': '-f json'}]}], parse_output=True)
+        assert out[0][dent]['rc'] == 0, 'Failed to get port detail of '
+        ports_state = [bond['state'] for bond in out[0][dent]['parsed_output']]
+        assert 'discarding' == ports_state[1], 'Bridge  does not have a blocking port'
+        await asyncio.sleep(convergence_time_s)
+
+        # 7. Verify traffic is forwarded  and port 2 doesn't receive any traffic
+        stats = await tgen_utils_get_traffic_stats(tgen_dev, 'Flow Statistics')
+        for row in stats.Rows:
+            if row['Traffic Item'] not in [traffic, traffic_1] and row['Rx Port'] == tg_ports[1]:
+                err_msg = f'Expected 0.0 got : {float(row["Rx Rate (Mbps)"])}'
+                assert isclose(float(row['Rx Rate (Mbps)']), 0.0, abs_tol=tolerance), err_msg
+            if row['Traffic Item'] in [traffic, traffic_1] and row['Rx Port'] != tg_ports[1]:
+                err_msg = f'Expected 300 got : {float(row["Rx Rate (Mbps)"])}'
+                assert isclose(float(row['Rx Rate (Mbps)']), 300, rel_tol=tolerance), err_msg
+
+        # 8. Randomly select between STP/RSTP modes
+        version = choice(['stp', 'rstp'])
```

Comment:
as discussed with @yuriilisovskyi - change the test steps 6-8 to be:
1) switch between STP-RSTP-STP
2) switch between RSTP-STP-RSTP
Choose either one or make the test parametrized and run both scenarios.